xds/internal/xdsclient/bootstrap/bootstrap.go

```diff
@@ -121,13 +148,16 @@ func (sc *ServerConfig) UnmarshalJSON(data []byte) error {
 	for _, cc := range xs.ChannelCreds {
 		// We stop at the first credential type that we support.
 		sc.CredsType = cc.Type
-		if cc.Type == credsGoogleDefault {
-			sc.Creds = grpc.WithCredentialsBundle(google.NewDefaultCredentials())
-			break
-		} else if cc.Type == credsInsecure {
-			sc.Creds = grpc.WithTransportCredentials(insecure.NewCredentials())
-			break
+		c := bootstrap.GetCredentials(cc.Type)
+		if c == nil {
+			continue
+		}
+		bundle, err := c.Build(cc.Config)
+		if err != nil {
+			return fmt.Errorf("failed to build credentials bundle from bootstrap for %q, error: %v", cc.Type, err)
```

Comment:
Nit: remove `, error`.  It's obvious what follows is the error message.

Reply:
Done